core/base/src/TApplication.cxx

```diff
@@ -628,20 +630,453 @@ TApplication::EExitOnException TApplication::ExitOnException(TApplication::EExit
+
+static TString URLforFunction(const TString &nameSpace, TClass *nameSpaceClass, const TString &functionName)
+{
+   // The URL starts as a URL for namespace only (generated with "UrlGenerator").
+   TString url(UrlGenerator(nameSpace, kURLforNameSpace));
+   // Then "#a" is appended to the URL.
+   url.Append("#a");
+   // Then text, crypted with MD5, is appended. The structure of the texst is as follows:
+   // "returnType nameSpace::functionNamefunctionName(function arguments)".
+   // "check" contains the argument of the function, changed with respect to Doxygen syntax.
+   TString check(FormatMethodForDoxygen(nameSpace, nameSpaceClass, functionName));
+   // The code below is temporary and must be removed when kIsConstexpr is available.
+   // In "TMath" in the master branch for example, a lot of the functions with no arguents
+   // are also "constexpr". In order to have them work for ".help" we will keep it like that.
+   // This provides problems for other functions (but not more than 5 in TMath).
+   TString returnType("");
+   if(check.EqualTo("()")){
+      // We will append "constexpr" to functions with no arguments.
+      returnType.Append("constexpr ");
+   }
+   // Can I use the FormatReturnTypeForDoxygen?
+   returnType.Append(((TFunction *)(nameSpaceClass->GetListOfMethods()->FindObject(functionName)))->GetReturnTypeName());
+   returnType.Append(" ");
+   // We folow the sequence: "returnType nameSpace::functionNamefunctionName(arguemnts)".
+   TString md5Text(returnType);
+   md5Text.Append(nameSpace);
+   md5Text.Append("::");
+   md5Text.Append(functionName);
+   md5Text.Append(functionName);
+   // We use "FormatMethodForDoxygen" (it works the same way for namespaces) to modify the arguments of 
+   // functionName with respect of Doxygen.
+   md5Text.Append(FormatMethodForDoxygen(nameSpace, nameSpaceClass, functionName));
+   // We crypt the text with MD5 - method in TString.
+   url.Append(md5Text.MD5());
+   return url;
+}
+}
+
+namespace {
+////////////////////////////////////////////////////////////////////////////////
+/// The function generates in a URL for "enumeratorName" defined in "className".
+/// It returns a TString with the URL used in the online reference guide,
+/// generated with Doxygen. The URL generation is specific. It could be
+/// separated into 3 parts - URL for "className", part for the enumeration
+/// and a part for the enumerator.
+///
+/// \param[in] className <the name of the class
+/// \param[in] enumeratorName <the name of the enumerator from the class
+
+static TString URLforEnum(const TString &className, TClass *classVariable, const TString &enumeratorName)
+{
+   // We create a TString with the name of the class and the enumeration from which "enumeratorName" is.
+   TString classEnumeration(((TDataMember *)(classVariable->GetListOfDataMembers()->FindObject(enumeratorName)))->GetTrueTypeName());
+   TString md5EnumClass("");
+   if ((classEnumeration.Contains("anonymous"))) {
```

Comment:
This catches `enum nonanonymousEnum { kOne }`. Can we be stricter than that, e.g. look for `(anonymous)`?